marketdata/stream/message.go

```diff
@@ -478,12 +478,17 @@ var errMessageHandler = func(c *client, e errorMessage) error {
 		c.pendingSubChange = nil
 	}
 
-	c.logger.Warnf("datav2stream: got error from server: %+v", e)
-	if e.code == 0 || e.msg == "" {
+	if e.code != 0 && e.msg != "" {
+		c.logger.Warnf("datav2stream: got error from server: %s", e)
+		return nil
+	}
+	if e.code == 0 && e.msg == "" {
+		return fmt.Errorf("datav2stream: received unexpected error: no code or msg")
```

Comment:
This should not be wrapped with "datav2stream: received unexpected error:", because the caller wraps it when logging. Same for the other branches.